### ISSUE TITLE
Questions around data preparation and hyperparameters for linear probing

### ISSUE THREAD

**Issue by RyanAmaudruz** [zhu-xlab/SSL4EO-S12#21]
Hi,
I first wanted to thank you for answering so promptly the issues in the repo.
I have been trying to reproduce some of your results. In particular, I have been trying to reproduce the linear probing results on the BE (100%) using your DINO ViT-S16 pretrained weights where you report an MAP of 83.4%.
Using the settings that are available in _src/benchmark/transfer_classification/scripts/benchmark/srun_lc_dino_rn50_s2c_BE.sh_ and the default values in _src/benchmark/transfer_classification/linear_BE_dino.py_, I am only able to obtain an MAP of just below 80%.
In the appendix of your paper, you report that you perform grid search to find the best hyperparams. Is it correct to assume that best hyperparams are the ones found in the default values of the python file and in the shell script? If not, would you have available the best hyperparams and would be willing to share them?

If the hyperparams are the same ones I used. The only difference I can imagine would be in the data preparation. Would it be possible to share how you prepared/scaled the BE dataset?

Thank you so much,
Ryan

**Comment by wangyi111** [comment]
Hi Ryan,

Thanks for your feedback! The provided script [srun_lc_dino_vits16_s2c_BE.sh](https://github.com/zhu-xlab/SSL4EO-S12/blob/main/src/benchmark/transfer_classification/scripts/benchmark/srun_lc_dino_vits16_s2c_BE.sh) is the one we used to get the reported results. Mainly a lr of 0.1 and batch size 64 per GPU (in total 256 in our machine). Therefore I guess the data preprocessing could be the reason. We generally follow the data preparation as [SeCo](https://github.com/ServiceNow/seasonal-contrast/blob/main/datasets/bigearthnet_dataset.py), our code is [bigearthnet_dataset_seco.py](https://github.com/zhu-xlab/SSL4EO-S12/blob/main/src/benchmark/transfer_classification/datasets/BigEarthNet/bigearthnet_dataset_seco.py). Specifically, after downloading the BigEarthNet-v1.0 data (590,326 images), we
- remove patches with too much snow and clouds (about 70K removed);
- normalize the data with mean and std, remove outlier, and convert to uint8;
- create lmdb files for train/val/test (311667/103944/103728 valid entries respectively).

Randomness could affect slightly the results but should be within ±1%. Here's the log of one run for example: https://drive.google.com/file/d/1T2TcqtMH5jYJHmqDi6Q5aeH1Q76ezwKf/view?usp=sharing.

\***************************

PS: I noticed an error in https://github.com/zhu-xlab/SSL4EO-S12/blob/74c53bfc6031d75a6b8f45068a3e5502e6872a55/src/benchmark/transfer_classification/linear_BE_dino.py#L83. For BigEarthNet the actual cropping scale is set as (0.8, 1.0) to be not too aggressive, as e.g. in https://github.com/zhu-xlab/SSL4EO-S12/blob/74c53bfc6031d75a6b8f45068a3e5502e6872a55/src/benchmark/transfer_classification/linear_BE_moco.py#L164.

This could affect slightly the results but also not much (<0.5%). I will correct this script. Thanks for the feedback to let me check!

**Comment by RyanAmaudruz** [comment]
Thank you so much for the prompt response and for providing the log. This will be very helpful in determining the gap between my result and yours.
Do you know why the log you shared has a training set of 300'000 and validation set of 100'000 images?
On my end, the raw training set is 354196 samples, minus 5599 snowy patches, minus 36930 cloudy patches, results in 311667 training samples. The raw validation set is 118065 samples, minus 1854 snowy patches, minus 12267 cloudy patches, results in 103944 validation samples.
This aligns with the sample count you mention in your comment.
I am thinking that if the additional 3944 validation samples are particularly hard to classify, this could explain the gap.
Do you know what those additional samples that were removed from the validation set are?
Thank you so much for your time

**Comment by RyanAmaudruz** [comment]
Ok, I have actually figured out why I was getting a lower result.
It was indeed the scaling of the data. I was scaling my data to have a mean of 0 and an std of 1, whilst you scaled the data to have a mean of 128 and an std of 64.
I am getting a similar train/val loss after changing the scaling.
Thank you so much for your help.

**Comment by RyanAmaudruz** [comment]
done
